npe2/manifest/utils.py

```diff
@@ -304,3 +324,211 @@ def merge_contributions(contribs: Sequence[Optional[ContributionPoints]]) -> dic
                             item["command"] = item["command"] + f"_{n + 2}"
             out = deep_update(out, c)
     return out
+
+
+def get_pypi_url(
+    name: str, version: Optional[str] = None, packagetype: Optional[str] = None
+) -> str:
+    """Get URL for a package on PyPI.
+
+    Parameters
+    ----------
+    name : str
+        package name
+    version : str, optional
+        package version, by default, latest version.
+    packagetype : str, optional
+        one of 'sdist', 'bdist_wheel', by default 'bdist_wheel' will be tried first,
+        then 'sdist'
+
+    Returns
+    -------
+    str
+        URL to download the package.
+
+    Raises
+    ------
+    ValueError
+        If packagetype is not one of 'sdist', 'bdist_wheel', or if version is specified
+        and does not match any available version.
+    KeyError
+        If packagetype is specified and no package of that type is available.
+    """
+    if packagetype not in {"sdist", "bdist_wheel", None}:
+        raise ValueError(
+            f"Invalid packagetype: {packagetype}, must be one of sdist, bdist_wheel"
+        )
+
+    with urlopen(f"https://pypi.org/pypi/{name}/json") as f:
+        data = json.load(f)
+
+    if version:
+        version = version.lstrip("v")
+        if version not in data["releases"]:
+            raise ValueError(f"{name} does not have version {version}")
+        _releases: List[dict] = data["releases"][version]
+    else:
+        _releases = data["urls"]
+
+    releases = {d.get("packagetype"): d for d in _releases}
+    if packagetype:
+        if packagetype not in releases:
+            version = version or "latest"
+            raise KeyError('No {packagetype} releases found for version "{version}"')
```

Comment:
```suggestion
            raise KeyError(f'No {packagetype} releases found for version "{version}"')
```

This should be an f-string no?